pkg/registration/hub/importer/providers/interface.go

```diff
@@ -0,0 +1,63 @@
+package providers
+
+import (
+	"context"
+
+	"github.com/openshift/library-go/pkg/controller/factory"
+	apiextensionsclient "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+
+	operatorclient "open-cluster-management.io/api/client/operator/clientset/versioned"
+	clusterv1 "open-cluster-management.io/api/cluster/v1"
+)
+
+// Interface is the interface that a cluster provider should implement
+type Interface interface {
+	// Clients is to return the client to connect to the target cluster.
```

Comment:
Should we add some comments about what permission the returned client should have?